doc/stdenv/platform-notes.chapter.md

```diff
@@ -140,81 +118,61 @@ Some common issues when packaging software for Darwin:
+| 16.0               | 15.0                                              | `apple-sdk_15`    |
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "foo-1.2.3";
-    # ...
-    buildInputs = [ apple-sdk_15 (darwinMinVersionHook "10.15") ]; # Upstream builds with the 15.0 SDK but supports 10.15.
-  }
-  ```
 
-- Libraries that require a minimum version can propagate an appropriate SDK and `darwinMinVersionHook`. Derivations using that library will automatically use an appropriate SDK and minimum version. Even if the library builds with a newer SDK, it should propagate the minimum supported SDK. Derivations that need a newer SDK can add it to their `buildInputs`.
+#### Darwin Default SDK versions
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3";
-    # ...
-    buildInputs = [ apple-sdk_15 ]; # Upstream builds with the 15.0 SDK but supports 10.15.
-    propagatedBuildInputs = [ apple-sdk_10_15 (darwinMinVersionHook "10.15") ];
-  }
-  # ...
-  stdenv.mkDerivation {
-    name = "bar-1.2.3";
-    # ...
-    buildInputs = [ libfoo ]; # Builds with the 10.15 SDK
-  }
-  # ...
-  stdenv.mkDerivation {
-    name = "baz-1.2.3";
-    # ...
-    buildInputs = [ apple-sdk_12 libfoo ]; # Builds with the 12.3 SDK
-  }
-  ```
+The current default versions of the deployment target (minimum version) and SDK are indicated by Darwin-specific attributes on the platform. Because of the ways that minimum version and SDK can be changed that are not visible to Nix, they should be treated as lower bounds. If you need to parameterize over a specific version, create a function that takes the version as a parameter instead of relying on these attributes.
 
-- Many SDK libraries and frameworks use text-based stubs to link against system libraries and frameworks, but several are built from source (typically corresponding to the source releases for the latest release of macOS). Several of these are propagated to your package automatically. They can be accessed via the `darwin` package set along with others that are not propagated by default.
+- `darwinMinVersion` defaults to 10.12 on x86_64-darwin and 11.0 on aarch64-darwin. It sets the default deployment target.
+- `darwinSdkVersion` defaults to 10.12 on x86-64-darwin and 11.0 on aarch64-darwin. Only the major version determines the SDK version, resulting in the 10.12.2 and 11.3 SDKs being used on these platforms respectively.
 
-  - libiconv
-  - libresolv
-  - libsbuf
+#### How to use libiconv on Darwin
 
-  Other common libraries are available in Darwin-specific versions with modifications from Apple. Note that these packages may be made the default on Darwin in the future.
+The libiconv package is included in the SDK by default along with libresolv and libsbuf. You do not need to do anything to use these packages. They are available automatically. If your derivation needs the `iconv` binary, add the `libiconv` package to your `nativeBuildInputs` (or `nativeCheckInputs` for tests).
```

Comment:
Not sure this section is necessary any more, as it is the assumption from the other platforms we support that `libiconv` is necessary. It could go under a more general “migration from the old pattern” heading along with the `apple_sdk` stuff, perhaps.

We should maybe have an earlier mention of things like CUPS that are part of the upstream SDK but need specifying explicitly with Nixpkgs though?

Reply:
There’s a lot of `if darwin, therefore libiconv` in nixpkgs. I want to address any shared knowledge that it’s necessary to continue doing that. If someone wants to remove it in a package, they can point to this section in the documentation if it’s questioned in review.

Reply:
Also, I added libutil to the documentation. If it makes more sense, I can split it out into the separate libutil commit that symlinks the dylib, but I’d like to leave it here for review until this is fully iterated and ready.